### PR TITLE
feat: create organization filter on home page

### PR DESCRIPTION
**Pull request by ruzniaievdm**
[2U-47](https://youtrack.raccoongang.com/issue/2U-47)
[2U-632](https://youtrack.raccoongang.com/issue/2U-632)
[2U-624](https://youtrack.raccoongang.com/issue/2U-624)

### CODE REVIEW

**Comment by ruzniaievdm** [comment]
@wowkalucky please merge commits separate

File: src/assets/scss/_form.scss [hunk 1]
```diff
@@ -79,3 +79,14 @@
     color: $black;
   }
 }
+
+.dropdown-group-wrapper {
+  position: relative;
+  z-index: 2;
```
Comment (PKulkoRaccoonGang) [inline src/assets/scss/_form.scss:85]: can we try to use some of the `z-index` that [Paragon](https://github.com/openedx/paragon/blob/master/scss/core/_variables.scss#L792) provides?

File: src/constants.js [hunk 2]
```diff
@@ -23,3 +23,14 @@ export const NOTIFICATION_MESSAGES = {
   duplicating: 'Duplicating',
   deleting: 'Deleting',
 };
+
+const COURSE_CREATOR_STATUSES = {
```
Comment (PKulkoRaccoonGang) [inline src/constants.js:27]: In the constants above, the export is done before the declaration of the constants. Let's follow this rule

File: src/generic/create-or-rerun-course/CreateOrRerunCourseForm.jsx [hunk 3]
```diff
@@ -204,21 +204,20 @@ const CreateOrRerunCourseForm = ({
...
-                    {option}
-                  </Form.AutosuggestOption>
-                ))}
-              </Form.Autosuggest>
+                handleBlur={handleCustomBlurForDropdown}
+                handleChange={(value) => setFieldValue(field.name, value)}
+                noOptionsMessage={intl.formatMessage(messages.courseOrgNoOptions)}
+                helpMessage=""
+                errorMessage=""
+                floatingLabel=""
```
Comment (PKulkoRaccoonGang) [inline src/generic/create-or-rerun-course/CreateOrRerunCourseForm.jsx:219]: Can we make optional props for this component? Add change to @edx/frontend-lib-content-components

File: src/studio-home/hooks.jsx [hunk 4]
```diff
@@ -1,39 +1,34 @@
...
 import { history } from '@edx/frontend-platform';
+import { useLocation } from 'react-router-dom';
 import { useDispatch, useSelector } from 'react-redux';
 
 import { RequestStatus } from '../data/constants';
-import {
-  updateSavingStatus,
-} from '../generic/data/slice';
+import { COURSE_CREATOR_STATUSES } from '../constants';
+import { updateSavingStatus } from '../generic/data/slice';
```
Comment (PKulkoRaccoonGang) [inline src/studio-home/hooks.jsx:8]:
```suggestion
import { updateSavingStatus } from '../generic/data/slice';
import { COURSE_CREATOR_STATUSES } from '../constants';
```

File: src/studio-home/hooks.jsx [hunk 5]
```diff
@@ -1,39 +1,34 @@
...
 import { getCourseData, getRedirectUrlObj, getSavingStatus } from '../generic/data/selectors';
+import { redirectToCourseIndex } from './constants';
+import { fetchStudioHomeData } from './data/thunks';
```
Comment (PKulkoRaccoonGang) [inline src/studio-home/hooks.jsx:11]:
```suggestion
import { fetchStudioHomeData } from './data/thunks';
import { redirectToCourseIndex } from './constants';
```

File: src/studio-home/organization-section/index.jsx [hunk 6]
```diff
@@ -1,34 +1,69 @@
...
+          handleBlur={(e) => setInputValue(e.target.value)}
+          handleChange={(value) => setInputValue(value)}
+          noOptionsMessage={intl.formatMessage(messages.organizationInputNoOptions)}
+          helpMessage=""
+          errorMessage=""
+          floatingLabel=""
         />
       </Form.Group>
-      <Button>
+      <Button onClick={() => handleSubmit()}>
```
Comment (PKulkoRaccoonGang) [inline src/studio-home/organization-section/index.jsx:66]:
```suggestion
      <Button onClick={handleSubmit}>
```

File: src/generic/create-or-rerun-course/CreateOrRerunCourseForm.jsx (hunk 3 above)
Comment (ruzniaievdm) [reply src/generic/create-or-rerun-course/CreateOrRerunCourseForm.jsx:219]: I think this task would take a lot of effort, it's better to leave it as it is.

File: src/studio-home/hooks.jsx (hunk 4 above)
Comment (ruzniaievdm) [reply src/studio-home/hooks.jsx:8]: what's the difference?

File: src/studio-home/hooks.jsx (hunk 5 above)
Comment (ruzniaievdm) [reply src/studio-home/hooks.jsx:11]: what's the difference?

File: src/studio-home/organization-section/index.jsx (hunk 6 above)
Comment (ruzniaievdm) [reply src/studio-home/organization-section/index.jsx:66]: sure, changed

**Review by PKulkoRaccoonGang — APPROVED** [review]

**Review by vladislavkeblysh — APPROVED** [review]

**Review by wowkalucky — APPROVED** [review]
OK 🤞

**Pull request merged**